common-test-lib/src/main/java/com/google/gct/idea/testing/TestUtils.java

```diff
@@ -127,6 +129,7 @@ public static MockProject mockProject(@Nullable PicoContainer container) {
     return new MockProject(container, getParentDisposableForCleanup());
   }
 
+  @SuppressFBWarnings(value="DM_STRING_CTOR", justification="Warning due to string concatenation")
```

Comment:
Something seems off, since the findbug is explicitly about using the String
constructor when the code examples use string concatenation, which I assume
is fine.

If we can't figure out why there's a mismatch we should definitely blanket
disable. I don't want to ban or explicitly mute every instance of String +
concatenation.

On Tue, Mar 8, 2016 at 2:11 PM, Chanseok Oh <notifications@github.com>
wrote:

> In
> common-test-lib/src/main/java/com/google/gct/idea/testing/TestUtils.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/481#discussion_r55410352>
> :
>
> > @@ -127,6 +129,7 @@ public static MockProject mockProject(@Nullable PicoContainer container) {
> >      return new MockProject(container, getParentDisposableForCleanup());
> >    }
> >
> > +  @SuppressFBWarnings(value="DM_STRING_CTOR", justification="Warning due to string concatenation")
>
> FindBugs warns about calling a String constructor like in a case of String
> s = new String("str") or SomeFnCall(String("str")). FindBugs suggests to
> use the string primitive directly. Unfortunately, this warning comes up
> where strings are concat'ed, e.g., a few lines below, you will see two
> occurrences of string concats.
>
> It's about negligible performance impact. Perhaps we may blanket disable
> and forget about this warning forever instead of annotating each method.
> What's your thought?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/481/files#r55410352>
> .
>